README.md

```diff
@@ -26,35 +26,120 @@ flutter pub add flame_behaviors
+### Behavior
 
-A behavior can have it's own `Component`s for adding extra functionality related to the behavior. It cannot, however, have its own `Behavior`s.
+A behavior is a component that defines how an entity behaves. It can be attached to an `Entity` 
+and handle a specific behavior for that entity. Behaviors can either be generic for any entity 
+or you can specify the specific type of entity it requires:
 
 ```dart
-// Define a custom behavior by extending `Behavior`.
-class CollisionBehavior extends Behavior with CollisionCallbacks {
-  CollisionBehavior(this._hitbox) : super(children: [_hitbox]);
+// Can be added to any type of Entity.
+class MyGenericBehavior extends Behavior {
+  ...
+}
+
+// Can only be added to MyEntity and subclasses of it.
+class MySpecificBehavior extends Behavior<MyEntity> {
+  ...
+}
+```
 
-  final RectangleHitbox _hitbox;
+Each behavior can have it's own `Component`s for adding extra functionality related to the 
+behavior. For instance a `TimerComponent` to have time based behavioral activity:
 
+```dart
+class MyBehavior extends Behavior {
   @override
-  Future<void> onLoad() async {...}
+  Future<void> onLoad() async {
+    await add(TimerComponent(period: 5, repeat: true, onTick: _onTick));
+  }
+
+  void _onTick() {
+    // Do something every 5 seconds.
+  }
+}
+```
 
+**Note**: A behavior, by definition, cannot have it's own `Behavior`s. 
+
+#### Input behaviors
+
+The `flame_behaviors` package also provides input behaviors, these behaviors are just a 
+sugarcoating over the existing Flame input mixins for components. These behaviors will 
+trigger when their parent entity is being interacted with by the user. So these events 
+are always relative to the parent entity.
+
+```dart
+class MyDraggableBehavior extends DraggableBehavior<MyEntity> {
   @override
-  @mustCallSuper
-  void onCollision(Set<Vector2> intersectionPoints, PositionComponent other) {...}
+  bool onDragUpdate(DragUpdateInfo info) {
+    // Do something on drag update event.
+    return super.onDragUpdate(info);
+  }
+}
 ```
 
-### Entity
+**Note**: You still need to add the corresponding input mixins to your game class, see the 
+[Flame input docs](https://docs.flame-engine.org/1.2.0/flame/inputs/inputs.html) for more 
+information.
+
+### Collision detection
+
+Flame comes with a powerful built-in [collision detection system](https://docs.flame-engine.org/1.2.0/flame/collision_detection.html) 
+but this API is not strongly typed. Components always get the colliding component as a 
+`PositionComponent` and developers need to manually check what type of class it exactly is. And 
+because `flame_behaviors` is all about enforcing a strongly typed API, it provide a special 
+behavior called `CollisionBehavior` that describe what type of entity it is interested for 
+collision. It does not however do any real collision detection, that is done by the 
+`PropagatingCollisionBehavior`.
```

Comment:
```suggestion
Flame comes with a powerful built-in [collision detection system](https://docs.flame-engine.org/1.2.0/flame/collision_detection.html), but this API is not strongly typed. Components always get the colliding component as a `PositionComponent` and developers need to manually check what type of class it is. 

`flame_behaviors` is all about enforcing a strongly typed API. It provides a special behavior called `CollisionBehavior` that describes what type of entity it will target for collision. It does not, however, do any real collision detection. That is done by the `PropagatingCollisionBehavior`.
```